block/pkg/datastore/aws/aws.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+    "github.com/aws/aws-sdk-go/aws/awserr"
+	awsec2 "github.com/aws/aws-sdk-go/service/ec2"
+	dscommon "github.com/opensds/multi-cloud/block/pkg/datastore/common"
+	pb "github.com/opensds/multi-cloud/block/proto"
+	"github.com/opensds/multi-cloud/block/pkg/model"
+	. "github.com/opensds/multi-cloud/s3/error"
+	log "github.com/sirupsen/logrus"
+)
+
+type AwsAdapter struct {
+	session *session.Session
+}
+
+// Create EBS volume
+func (ad *AwsAdapter) Create(ctx context.Context, volume *model.Volume) (*pb.CreateVolumeResponse, error){
+    var result pb.CreateVolumeResponse
+    var tags []*awsec2.Tag
+
+    tagList := &awsec2.TagSpecification{
+        Tags:   append(tags, &awsec2.Tag{
+                        Key:   aws.String("Name"),
+                        Value: aws.String(volume.Name)}),
+        ResourceType: aws.String(awsec2.ResourceTypeVolume),
+    }
+
+    volCreationInput := &awsec2.CreateVolumeInput{
+        AvailabilityZone:   aws.String(volume.AvailabilityZone),
+        Size:               aws.Int64(volume.Size),
+        VolumeType:         aws.String(volume.Type),
+        TagSpecifications:   []*awsec2.TagSpecification{tagList},
+    }
+    if (volume.Type == "io1") {
+        // Iops is required only for io1 volumes
+        volCreationInput.Iops = aws.Int64(volume.Iops)
+    }
+	// Create a session of EC2
```

Comment:
What's the relathionship between our blockservice and EC2?

Reply:
AWS EBS is EC2 service unlike EFS